README.md

```diff
@@ -181,6 +181,112 @@ If you want to compile with ROCm support, install
+### Step 2: Set Up Downloads Directory (Optional)
+Create a symbolic link to your Downloads directory if you have one on your Windows system.
+```bash
+ln -s /mnt/c/Users/<user>/Downloads Downloads
+```
+Replace `<user>` with your Windows username.
+### Step 3: Install Anaconda
+Download Anaconda and install it. Make sure to give execute permissions to the downloaded Anaconda script.
+```bash
+wget https://repo.anaconda.com/archive/Anaconda3-2024.02-1-Linux-x86_64.sh
+chmod +x /home/<user>/Downloads/Anaconda3-2024.02-1-Linux-x86_64.sh
+./home/<user>/Downloads/Anaconda3-2024.02-1-Linux-x86_64.sh
+```
+Activate the newly installed Anaconda environment and create a new environment for PyTorch development.
+```bash
+exec bash
+conda create --name pytorch-dev python=3.12
+conda activate pytorch-dev
+```
+Install necessary Python packages for PyTorch.
+```bash
+conda install astunparse numpy ninja pyyaml mkl mkl-include setuptools cmake cffi typing_extensions future six requests dataclasses
+```
+### Step 4: Install NVIDIA CUDA Toolkit
+Install CUDA on your Windows machine from the 12.4.1 repo. Then, download and install the NVIDIA CUDA toolkit on Ubuntu.
+```bash
+wget https://developer.download.nvidia.com/compute/cuda/repos/wsl-ubuntu/x86_64/cuda-keyring_1.1-1_all.deb
+sudo dpkg -i cuda-keyring_1.1-1_all.deb
+sudo apt-get update
+sudo apt-get -y install cuda
+```
+### Step 5: Set CUDA Paths
+Set the CUDA paths to enable CUDA in your system. I've added these lines to my `.bashrc` file, so they persist after a reboot see below instuctions.
+```bash
+export PATH=/usr/local/cuda/bin/:$PATH
+export LD_LIBRARY_PATH=/usr/local/cuda/lib64:$LD_LIBRARY_PATH
+```
+Verify the CUDA installation. If this line does not give an output, it means the paths are not set correctly from the export commands above. You may not have linked the binaries with the export commands or updated the source to reflect the changes in the `.bashrc` file. A simple reboot may solve it.
+```bash
+nvcc --version
+```
+### Step 6: Install cuDNN
+Download and install cuDNN from the NVIDIA repository.
+```bash
+wget https://developer.download.nvidia.com/compute/cudnn/9.1.0/local_installers/cudnn-local-repo-ubuntu2004-9.1.0_1.0-1_amd64.deb
+sudo dpkg -i cudnn-local-repo-ubuntu2004-9.1.0_1.0-1_amd64.deb
+sudo cp /var/cudnn-local-repo-ubuntu2004-9.1.0/cudnn-*-keyring.gpg /usr/share/keyrings/
+sudo apt-get update
+sudo apt-get -y install cudnn-cuda-12
+```
+### Step 7: Update `.bashrc` for Permanent Paths
+Add the CUDA paths to your `~/.bashrc` file to make them permanent.
+```bash
+nano ~/.bashrc
+```
+Add the following lines at the end of the file:
+```bash
+export PATH=/usr/local/cuda/bin/:$PATH
+export LD_LIBRARY_PATH=/usr/local/cuda/lib64:$LD_LIBRARY_PATH
+```
+Save and close the file: `Ctrl + X`, then `Y`, then `Enter`. Then, reload the `~/.bashrc` file.
+```bash
+source ~/.bashrc
+```
+### Step 8: Reboot Your System
+Reboot your system to apply all the changes.
+```bash
+sudo reboot
+```
```

Comment:
This steps is unnecessary isn't it?

Reply:
This is in WSL. So sudo reboot is really fast. 1min tops And then all the paths are updated etc. I personally thinks it's good to do a reboot. Yet is it necessary like on working server or something perhaps not.

Reply:
i will update to this "exec bash"